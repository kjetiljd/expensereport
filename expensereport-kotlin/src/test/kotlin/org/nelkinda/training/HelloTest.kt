package org.nelkinda.training

import com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemOut
import org.approvaltests.Approvals
import org.junit.Ignore
import org.junit.Test
import kotlin.test.assertTrue

class HelloTest {

    @Test
    fun `first line in report has correct format`() {
        val noExpenses = emptyList<Expense>()

        val actual = tapSystemOut {
            ExpenseReport().printReport(noExpenses)
        }

        val expected = """Expenses \w{3} \w{3} \d{1,2} \d\d:\d\d:\d\d [A-Z]{3,4} 20\d\d""".toRegex()
        assertTrue(actual.lines().first().matches(expected))

    }

    @Test
    fun `golden master test of main program output`() {
        val noExpenses = emptyList<Expense>()

        val actual = tapSystemOut {
            ExpenseReport().printReport(noExpenses)
        }

        val actualWithoutDynamicHeader = actual.lines()
            .filterIndexed { index, _ -> index != 0 }
            .joinToString("\n")
        Approvals.verify(actualWithoutDynamicHeader)
    }

}
