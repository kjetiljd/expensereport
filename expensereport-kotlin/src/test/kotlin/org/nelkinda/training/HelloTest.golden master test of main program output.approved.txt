Meal expenses: 0
Total expenses: 0
